developer/reference/articles/tooling/blade-cli/08-extending-blade-cli/03-creating-custom-project-templates-for-blade-cli.markdown

```diff
@@ -0,0 +1,80 @@
+---
+header-id: creating-custom-project-templates-for-blade-cli
+---
+
+# Creating Custom Project Templates for Blade CLI
+
+[TOC levels=1-4]
+
+Blade comes with over 32+ project templates, but many times you may feel that
+those are too simple or don't fit the need for your development team. You can
+create new custom project templates that fit your team's workflow and have Blade
+use it instead.
+
+The setup of a custom project template should mimic that of a Maven archetype.
+The best way to illustrate this is by visualizing a
+[sample template](https://github.com/liferay/liferay-blade-cli/tree/master/extensions/sample-template)'s
+folder structure:
+
+- `src/`
+    - `main/resources/`
+        - `META-INF`
+            - `maven`
+                - `archetype-metadata.xml`
+            - `archetype-post-generate.groovy`
+        - `archetype-resources`
+            - Folder structure to be generated
+- `bnd.bnd`
+- `[build.gradle|pom.xml]`
+
+You can read more about Maven archetypes and their features and capabilities
+[here](https://maven.apache.org/guides/introduction/introduction-to-archetypes.html).
+
+To create a custom project template that can be generated using Blade CLI,
+follow these steps:
+
+1.  Create a generic Maven archetype following the folder structure outlined
+    above. Follow Maven's documentation to configure the archetype project
+    appropriately.
+
+2.  Open the template's `bnd.bnd` file and ensure it sets the following
+    configurations:
+
+    ```
+    Bundle-Description: TEMPLATE_DESCRIPTION
+    Bundle-Name: TEMPLATE_NAME
+    Bundle-SymbolicName: SYMBOLIC_NAME
+    Bundle-Version: TEMPLATE_VERSION
+    Liferay-Versions: LIFERAY_VERSION_RANGE
```

Comment:
Should we provide a link here to read more about bnd versioning and version ranges? 
Maybe something like this? https://bnd.bndtools.org/chapters/170-versioning.html

Reply:
sure; that would be good to reference